src/Lykke.Job.CandlesHistoryWriter.Services/HistoryMigration/TradesMigrationService.cs

```diff
@@ -140,52 +95,133 @@ TimeSpan sqlTimeout
 
             _tradesMigrationHealthService.State = TradesMigrationState.Finished;
         }
+        
+        public class TradesProcessor
+        {
+            private readonly TradesMigrationHealthService _healthService;
+            private readonly ICandlesHistoryRepository _historyRepo;
+            private readonly Dictionary<CandleTimeInterval, ICandle> _activeCandles;
+            private readonly string _assetPairId;
+            private readonly int _persistenceQueueMaxSize;
 
-        #endregion
+            // ReSharper disable once MemberCanBePrivate.Global
+            public readonly Dictionary<CandleTimeInterval, List<ICandle>> PersistenceCandleQueue; // It's better to make it private, but we need to unit test the class.
 
-        #region Private
+            public TradesProcessor(TradesMigrationHealthService healthService, ICandlesHistoryRepository historyRepo, string assetPairId, int persistenceQueueMaxSize)
+            {
+                _healthService = healthService ?? throw new ArgumentNullException(nameof(healthService));
+                _historyRepo = historyRepo ?? throw new ArgumentNullException(nameof(_historyRepo));
+                _assetPairId = !string.IsNullOrWhiteSpace(assetPairId)
+                    ? assetPairId
+                    : throw new ArgumentNullException(nameof(assetPairId));
+                _persistenceQueueMaxSize = persistenceQueueMaxSize;
+
+                PersistenceCandleQueue = new Dictionary<CandleTimeInterval, List<ICandle>>();
+                foreach (var si in Candles.Constants.StoredIntervals)
+                    PersistenceCandleQueue.Add(si, new List<ICandle>());
```

Comment:
You can use persistenceQueueMaxSize to initialize list capacity. This will reduce amount of memory allocations and underlying array copying.